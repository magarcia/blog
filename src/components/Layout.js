import React from 'react';
import { Link } from 'gatsby';
import DarkLightMode from './DarkLightMode';
import '../styles/index.css';

class Layout extends React.Component {
  renderHeader() {
    const { location, title } = this.props;
    const rootPath = `${__PATH_PREFIX__}/`; // eslint-disable-line no-undef

    if (location.pathname === rootPath) {
      return (
        <>
          <h1>
            <Link to="/">{title}</Link>
          </h1>
          <DarkLightMode />
        </>
      );
    }
    return (
      <>
        <h3>
          <Link to="/">{title}</Link>
        </h3>
        <DarkLightMode />
      </>
    );
  }

  render() {
    const { children } = this.props;

    return (
      <>
        <header className="header">{this.renderHeader()}</header>
        {children}
      </>
    );
  }
}

export default Layout;
