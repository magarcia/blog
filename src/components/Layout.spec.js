import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './Layout';

it('renders correctly root path', () => {
  const props = {
    location: {
      pathname: `${__PATH_PREFIX__}/`, // eslint-disable-line no-undef
    },
    title: 'title',
  };
  const tree = renderer
    .create(
      <Layout {...props}>
        <div>content</div>
      </Layout>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly non root path', () => {
  const props = {
    location: {
      pathname: `${__PATH_PREFIX__}/post`, // eslint-disable-line no-undef
    },
    title: 'title',
  };
  const tree = renderer
    .create(
      <Layout {...props}>
        <div>content</div>
      </Layout>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
