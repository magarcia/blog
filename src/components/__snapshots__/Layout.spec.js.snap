// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly non root path 1`] = `
Array [
  <header
    className="header"
  >
    <h3>
      <a
        href="/"
      >
        title
      </a>
    </h3>
    <div
      className="darkLightMode "
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
    >
      <svg
        className="feather feather-sun"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <title />
        <g
          fill="none"
          strokeLinecap="round"
          strokeWidth="2"
        >
          <circle
            cx="12"
            cy="12"
            r="5"
          />
          <path
            d="M12 1v2"
          />
          <path
            d="M12 21v2"
          />
          <path
            d="M4.22 4.22l1.42 1.42"
          />
          <path
            d="M18.36 18.36l1.42 1.42"
          />
          <path
            d="M1 12h2"
          />
          <path
            d="M21 12h2"
          />
          <path
            d="M4.22 19.78l1.42-1.42"
          />
          <path
            d="M18.36 5.64l1.42-1.42"
          />
        </g>
      </svg>
      <input
        aria-label="Switch between Dark and Light mode"
        checked={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
    </div>
  </header>,
  <div>
    content
  </div>,
]
`;

exports[`renders correctly root path 1`] = `
Array [
  <header
    className="header"
  >
    <h1>
      <a
        href="/"
      >
        title
      </a>
    </h1>
    <div
      className="darkLightMode "
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
    >
      <svg
        className="feather feather-sun"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <title />
        <g
          fill="none"
          strokeLinecap="round"
          strokeWidth="2"
        >
          <circle
            cx="12"
            cy="12"
            r="5"
          />
          <path
            d="M12 1v2"
          />
          <path
            d="M12 21v2"
          />
          <path
            d="M4.22 4.22l1.42 1.42"
          />
          <path
            d="M18.36 18.36l1.42 1.42"
          />
          <path
            d="M1 12h2"
          />
          <path
            d="M21 12h2"
          />
          <path
            d="M4.22 19.78l1.42-1.42"
          />
          <path
            d="M18.36 5.64l1.42-1.42"
          />
        </g>
      </svg>
      <input
        aria-label="Switch between Dark and Light mode"
        checked={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
    </div>
  </header>,
  <div>
    content
  </div>,
]
`;
